### DISCUSSION TITLE
How to replace text with regular expression?

### DISCUSSION THREAD

**Discussion by supply9243** [hluk/CopyQ#2530] [Q&A]
The text I'm going to copy
magnet:?xt=urn:btih:fc0a90cb47251361d696fd2057e82d3f3ee2b108&dn=zh-cn_windows_10_business_editions_version_22h2_updated_oct_2023_x64_dvd_eb713365.iso&xl=6467465216

I want it to show up inside CopyQ as
magnet:?xt=urn:btih:fc0a90cb47251361d696fd2057e82d3f3ee2b108

I can use regular matching, but I don't know how to replace it. Please help.😵
`^(magnet:\?xt=urn:btih:\w{40}).+`

**Comment by ooker777** [comment]
Try this:
```js
copyq:
var raw = str(clipboard())
var processed = raw.replace(new RegExp("matched text", "g"), "replaced text")
copy(processed)
paste()  
```

**Reply by supply9243** [reply]
Thanks for your answer, but I don't understand why it doesn't work after modifying it.
```
copyq:
var raw = str(clipboard())
var processed = raw.replace(new RegExp("^magnet:\?xt=urn:btih:(\w{40}).+", "g"), "magnet:\?xt=urn:btih:$1")
copy(processed)
paste()  
```

**Comment by supply9243** [comment]
I found a piece of code that wasn't intended for this, but modified it and it works.
```
copyq:
var text = str(input())
if ( text.indexOf('magnet') == -1 )
    abort()

var newText = text.replace(/^magnet:\?xt=urn:btih:(\w{40}).+/g, 'magnet:\?xt=urn:btih:$1')
setData(mimeText, newText)
removeData(mimeHtml)
copy(newText)

```
